### PR TITLE
Fix "reg vs new-reg" encoding problem.

### PR DESCRIPTION
**Pull request by kuba**
Unfortunately, I couldn't find an optimal solution, but it at least unblocks Boulder (https://github.com/letsencrypt/boulder/pull/442)...

Builds on top of #618.

### CODE REVIEW

**Comment by kuba** [comment]
FYI this is successfully tested (`./tests/boulder-integration.sh`) against #618.

File: acme/acme/messages.py [hunk 1]
```diff
@@ -305,6 +308,10 @@ def resolved_combinations(self):
         return tuple(tuple(self.challenges[idx] for idx in combo)
                      for combo in self.combinations)
 
+class NewAuthorization(Authorization):
+    """New authorization."""
+    resource_type = 'new-authz'
+    resource = fields.Resource(resource_type)
```
Comment (jdkasten) [inline acme/acme/messages.py]: All of the Authorization attributes are proxied... right?

Comment (kuba) [inline acme/acme/messages.py]: They are inherited rather than proxied, but yes.

Comment (jdkasten) [inline acme/acme/messages.py]: Ahh... sorry yes.  Great!

**Comment by jdkasten** [comment]
Thanks @kuba!

**Pull request merged**